@xen-orchestra/fs/src/smb.js

```diff
@@ -168,4 +168,10 @@ export default class SmbHandler extends RemoteHandlerAbstract {
       .writeFile(this._getFilePath(file), data, options)
       .catch(normalizeError)
   }
+
+  _truncate(file, len) {
+    return this._client
+      .truncate(this._getFilePath(file), len)
```

Comment:
This is not available in the library we are using ([`@marsaud/smb2`](https://github.com/Node-SMB/marsaud-smb2/)), we'll need to create it.